packages/x-data-grid-premium/src/tests/clipboard.DataGridPremium.test.tsx

```diff
@@ -59,20 +61,14 @@ describe('<DataGridPremium /> - Clipboard', () => {
   }
 
   describe('copy', () => {
-    let writeText: SinonStub;
-    const originalClipboard = navigator.clipboard;
+    let writeText: SinonSpy;
 
     beforeEach(function beforeEachHook() {
-      writeText = stub().resolves();
-
-      Object.defineProperty(navigator, 'clipboard', {
```

Comment:
With the latest version of `@mui/internal-test-utils` we are using `@testing-library/user-event`.
`userEvent.setup` is called in the render function and defines `navigator.clipboard`.

I decided to export `rtlUserEvent` from `@mui/internal-test-utils` to have the option to setup the mocks sooner and have the option to spy on them.